File: androidApp/src/main/java/com/dev/noteappkmm/android/note_list/component/NoteItem.kt
package com.dev.noteappkmm.android.note_list.component

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.interaction.MutableInteractionSource
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.Icon
import androidx.compose.material.MaterialTheme
import androidx.compose.material.Text
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Close
import androidx.compose.runtime.Composable
import androidx.compose.runtime.remember
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.dev.noteappkmm.domain.models.Note
import com.dev.noteappkmm.domain.time.DateTimeUtil

@Composable
fun NoteItem(
    modifier: Modifier = Modifier,
    note: Note,
    backgroundColor: Color,
    onNoteClick: () -> Unit,
    onDeleteClick: () -> Unit
) {
    val formattedDate = remember(note.created) {
        DateTimeUtil.formatNoteDate(note.created)
    }

    Column(
        modifier = modifier
            .clip(RoundedCornerShape(5.dp))
            .background(backgroundColor)
            .clickable { onNoteClick() }
            .padding(16.dp)
    ) {
        Row(
            modifier = Modifier
                .fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically
        ) {
            Text(
                text = note.title,
                style = MaterialTheme.typography.subtitle2.copy(
                    fontSize = 20.sp,
                    fontWeight = FontWeight.SemiBold
                )
            )

            Icon(
                modifier = Modifier
                    .clickable(
                        MutableInteractionSource(),
                        null
                    ) { onDeleteClick() },
                imageVector = Icons.Default.Close,
                contentDescription = "Delete Note"
            )
        }
        
        Spacer(modifier = Modifier.height(16.dp))
        
        Text(
            text = note.content,
            style = MaterialTheme.typography.caption.copy(
                fontWeight = FontWeight.Light
            )
        )

        Spacer(modifier = Modifier.height(16.dp))

        Text(
            modifier = Modifier
                .align(Alignment.End),
            text = formattedDate,
            color = Color.DarkGray
        )
    }
}

File: iosApp/iosApp/note_list/component/NoteItem.swift
import SwiftUI
import shared

struct NoteItem: View {
    var note: Note
    let onDeleteClick: () -> Void
    
    var body: some View {
        VStack(alignment: .leading) {
            HStack {
                Text(note.title)
                    .font(.title3)
                    .fontWeight(.semibold)
                
                Spacer()
                
                Image(systemName: "xmark")
                    .onTapGesture {
                        onDeleteClick()
                    }
            }.padding(.bottom, 4)
            
            Text(note.content)
                .fontWeight(.light)
                .padding(.bottom, 4)
            
            HStack {
                Spacer()
                
                Text(
                    DateTimeUtil()
                        .formatNoteDate(
                            dateTime: note.created
                        )
                )
                .font(.footnote)
                .fontWeight(.light)
            }
        }
        .padding()
        .background(Color(hex: note.colorHex))
        .clipShape(RoundedRectangle(cornerRadius: 4.0))
    }
}

struct NoteItem_Previews: PreviewProvider {
    static var previews: some View {
        NoteItem(
            note: Note(
                id: nil,
                title: "My Note",
                content: "My Content",
                colorHex: 0xFF2341,
                created: DateTimeUtil().now()
            ),
            onDeleteClick: {}
        )
    }
}
